### PR TITLE
Feature/refactor dbnsfp cadd pipeline

### PR DESCRIPTION
**Pull request by at7**
This will be in production for release 98. We will discuss the pipeline in our Monday meeting. Confluence documentation is in progress.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Variation/Pipeline/BaseVariationProcess.pm [hunk 1]
```diff
@@ -149,5 +149,28 @@ sub run_cmd {
   }
 }
 
+sub get_assembly {
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Pipeline/BaseVariationProcess.pm]: Bio::EnsEMBL::DBSQL::GenomeContainer::get_version() can be used to get assembly version

File: modules/Bio/EnsEMBL/Variation/Pipeline/BaseVariationProcess.pm [hunk 2]
```diff
@@ -149,5 +149,28 @@ sub run_cmd {
...
+  my $assembly;
+  $sth->bind_columns(\$assembly);
+  $sth->execute();
+  $sth->fetch();
+  $sth->finish();
+
+  return $assembly;
+}
+
+sub get_species_id {
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Pipeline/BaseVariationProcess.pm]: Bio::EnsEMBL::DBSQL::MetaContainer method species_id can be used to get species_id

File: modules/Bio/EnsEMBL/Variation/Utils/DbNSFPProteinFunctionAnnotation.pm [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+package Bio::EnsEMBL::Variation::Utils::DbNSFPProteinFunctionAnnotation;
+
+use Bio::EnsEMBL::Variation::Utils::BaseProteinFunctionAnnotation;
+our @ISA = ('Bio::EnsEMBL::Variation::Utils::BaseProteinFunctionAnnotation');
+
+my $REVEL_CUTOFF = 0.5;
+
+=head2 new
+  Example    :
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Utils/DbNSFPProteinFunctionAnnotation.pm]: perldoc DbNSFPProteinFunctionAnnotation.pm does not display clearly. Adding blank line after =head2 helps.
Similarly for get_dbNSFP_row  description

File: modules/Bio/EnsEMBL/Variation/Utils/DbNSFPProteinFunctionAnnotation.pm [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  my $cadd = Bio::EnsEMBL::Variation::Utils::DbNSFPProteinFunctionAnnotation->new(
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Utils/DbNSFPProteinFunctionAnnotation.pm]: Variable is $cadd, should it be a  $dbsnp to be clearer

File: modules/Bio/EnsEMBL/Variation/Utils/BaseDatabaseUtils.pm [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  my ($species, $registry_file) = rearrange([qw(SPECIES REGISTRY_FILE)], @_);
+  my $self = bless {
+      'species' => $species,
+      'registry_file' => $registry_file,
+  }, $class;
+
+  return $self;
+}
+
+sub adaptor {
```
[the remark this answers is not in the document]
Comment (at7) [reply modules/Bio/EnsEMBL/Variation/Utils/BaseDatabaseUtils.pm]: It didn't need the adaptor in the end and have removed it now

File: modules/Bio/EnsEMBL/Variation/Utils/CADDProteinFunctionAnnotation.pm [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  Description: Join header column with row value
+  Returntype : Hashref mapping header column to row value
+  Exceptions : None
+  Caller     : load_predictions_for_triplets() 
+  Status     : 
+=cut
+sub get_CADD_row {
+  my $self = shift;
+  my $line = shift;
+  $line =~ s/\r$//g;
```
[the remark this answers is not in the document]
Comment (at7) [reply modules/Bio/EnsEMBL/Variation/Utils/CADDProteinFunctionAnnotation.pm]: I cannot remember where this i coming from and why it is needed. I removed it and the tests are still passing.

File: modules/Bio/EnsEMBL/Variation/Utils/BaseDatabaseUtils.pm [hunk 7]
```diff
@@ -0,0 +1,108 @@
+=head1 LICENSE
```
[the remark this answers is not in the document]
Comment (at7) [reply modules/Bio/EnsEMBL/Variation/Utils/BaseDatabaseUtils.pm]: I wanted to have the code under Bio/EnsEMBL/Variation/Utils separate from the code under Bio/EnsEMBL/Variation/Pipeline. BaseVariationProcess.pm inherits from Bio::EnsEMBL::Hive::Process and I don't know if that is going to cause conflicts. I also find the name BaseVariationProcess confusing. The idea is that you can run CADDProteinFunctionAnnotation.pm outside of ehive pipelines too. But the structure of the modules and where we should keep them under Utils or Pipeline can be discussed separately and I will add a comment to the confluence page.

**Comment by at7** [comment]
Thank you Helen for your feedback. I implemented all your suggestions except for where to store the modules. I leave them under Utils for now. We do use the Utils directory for other ehive pipelines too and can discuss if they should be moved under Pipeline in the future separately.

**Pull request merged**